yt_idefix/data_structures.py

```diff
@@ -427,15 +427,175 @@ def _get_time(self):
 
     def _set_code_unit_attributes(self):
         """Conversion between physical units and code units."""
-        length_unit = self.parameters.get("unit_length", 1.0)
-        velocity_unit = self.parameters.get("unit_velocity", 1.0)
-        density_unit = self.parameters.get("unit_density", 1.0)
 
-        self.length_unit = self.quan(length_unit, "cm")
-        self.velocity_unit = self.quan(velocity_unit, "cm/s")
-        density_unit = self.quan(density_unit, "g/cm**3")
+        # Overriding units are preferred to PLUTO's base units
+        pluto_units = {
+            "length_unit": "cm",
+            "velocity_unit": "cm/s",
+            "density_unit": "g/cm**3",
+        }
+        spu = set(pluto_units)
+        if len(self.units_override) < 3:
+            if len(self.units_override):
```

Comment:
equivalent but clearer (in my opinion)
```suggestion
            if len(self.units_override) > 0:
```